cparser/Elab.ml

```diff
@@ -1926,38 +1983,42 @@ let __func__type_and_init s =
 
 let enter_typedefs loc env sto dl =
   if sto <> Storage_default then
-    error loc "Non-default storage on 'typedef' definition";
+    error loc "cannot combine with previous 'typedef' declaration specifier";
   List.fold_left (fun env (s, ty, init) ->
     if init <> NO_INIT then
       error loc "initializer in typedef";
     match previous_def Env.lookup_typedef env s with
     | Some (s',ty') ->
-        if equal_types env ty ty' then begin
-          warning loc "redefinition of typedef '%s'" s;
```

Comment:
Redefinition of a typedef with the same type (e.g. `typedef int t; typedef int t;`) is ISO C11 but not ISO C99.  `clang -std=c99` and `gcc -std=c99 -pedantic` warn about it.